test/framework/prometheus.go

```diff
@@ -305,6 +305,42 @@ func (f *Framework) MakeThanosQuerierService(name string) *v1.Service {
 	return service
 }
 
+// CreatePrometheusAndWaitUntilReadyV055x create and wait prometheus for operator v0.55.x.
+// This is needed for the upgrade path e2e test from v0.55.x to v0.56.x.
+// It can be deprecated when the latest version reach 0.57.x.
```

Comment:
@heylongdacoder would you mind adding additional info to the comment as to why it is needed and perhaps a `#TODO` so it does not get lost.

Reply:
@PhilipGough done, please check if the additional info is clear enough. I also added another #TODO [here](https://github.com/prometheus-operator/prometheus-operator/pull/4758/files#diff-4049027b7623040b566fa582bcbf90a0c469204019a56d366134c5090720c0afR168)